### DISCUSSION TITLE
"Select All" or "Unselect All" filter option

### DISCUSSION THREAD

**Discussion by arindammitra06** [KevinVandy/mantine-react-table#351] [Q&A]
### mantine-react-table version

1.3.4

### react & react-dom versions

18.2.0

### Describe the bug and the steps to reproduce it

First of all, thank you for the awesome library.
I am using Mantive Table v6 in one of my project and very much love its features. I am using the 'multi-select' filter variant for my tables, I was wondering if there is a way to have the "Select All" or "Unselect All" options in the filter selection. Does the enableSelectAll help anyhow, i guess its for the row selection.

If not available, this should rather go as an enhancement request. Sorry for the trouble and thank You in advance. 

### Minimal, Reproducible Example - (Optional, but Recommended)

const table = useMantineReactTable({
        columns,
        data,
        layoutMode:'semantic',
        **_enableSelectAll: true,_**
        selectAllMode:'all',
        enableColumnResizing: true,
        enableColumnPinning:true,
        enableColumnDragging:true,
        enableColumnOrdering:true,
        enableColumnFilterModes: true,
        enableFacetedValues: true,
        enableGrouping: true,
        columnFilterDisplayMode: 'popover',
        paginationDisplayMode: 'pages',
        positionToolbarAlertBanner: 'bottom',
        enableStickyHeader: true,
        mantineTableContainerProps: { sx: { maxHeight: '350px'} },
        mantineSearchTextInputProps: {
            placeholder: `Search ${data.length} rows`,
            size:'xs'
            
        },
        mantineTableBodyCellProps:{
            sx: {
                fontSize:'12px !important',
            }
        },
        mantineTableHeadCellProps:{
            sx: {
                fontSize:'13px !important',
            }
        },
        mantinePaginationProps: {
            radius: 'md',
            size: 'xs',
        },
        mantineTableProps:{
            striped: true,
        },
        mantineFilterMultiSelectProps:{
            radius: 'xs',
            size: 'sm',
        },
        initialState: {
            density: 'xs',
            expanded: true,
            pagination: { pageIndex: 0, pageSize: 20 },
            showColumnFilters: true, 
            showGlobalFilter: true
        },
        state: { isLoading: isLoading },
      });

### Do you intend to try to help solve this bug with your own PR?

Maybe, I'll investigate and start debugging

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by alessandrojcm** [comment]
You can easily achieve this using `mantineFilterMultiSelectProps`, just set it as 'clearable', pass the "select all" as first element of your data array and then implement the change handler, see: https://codesandbox.io/p/devbox/optimistic-golick-547ymw?file=%2Fsrc%2FTS.tsx%3A86%2C1

**Reply by arindammitra06** [reply]
Thats awesome. Thank You. I was wondering that there would be any way to have the filter selection like the below picture, that wouldn't require us passing filter options as array into mantineFilterMultiSelectProps, rather would be populated automatically by passing filterVariant: 'multi-select'.

Thanks again.
